RiotSwiftUI/Modules/UserSessions/UserSessionsOverview/View/UserSessionsOverview.swift

```diff
@@ -127,23 +127,26 @@ struct UserSessionsOverview: View {
                 Button { viewModel.send(viewAction: .renameCurrentSession) } label: {
                     Label(VectorL10n.manageSessionRename, systemImage: "pencil")
                 }
-            }
-            
-            if #available(iOS 15, *) {
-                Button(role: .destructive) { viewModel.send(viewAction: .logoutOfCurrentSession) } label: {
-                    Label(VectorL10n.signOut, systemImage: "rectangle.portrait.and.arrow.right.fill")
-                }
-            } else {
-                Button { viewModel.send(viewAction: .logoutOfCurrentSession) } label: {
-                    Label(VectorL10n.signOut, systemImage: "rectangle.righthalf.inset.fill.arrow.right")
+                if #available(iOS 15, *) {
+                    Button(role: .destructive) { viewModel.send(viewAction: .logoutOfCurrentSession) } label: {
+                        Label(VectorL10n.signOut, systemImage: "rectangle.portrait.and.arrow.right.fill")
+                    }
+                } else {
+                    Button { viewModel.send(viewAction: .logoutOfCurrentSession) } label: {
+                        Label(VectorL10n.signOut, systemImage: "rectangle.righthalf.inset.fill.arrow.right")
+                    }
```

Comment:
Since I saw this pattern many times around, I was wondering if we can save some code by defining a modifier or a Button.init to tackle this specific problem.
If you like the init idea it would be something like that:

```swift
extension Button {
    init(destructiveAction: @escaping () -> Void, @ViewBuilder label: () -> Label) {
        if #available(iOS 15, *) {
            self.init(role: .destructive, action: destructiveAction, label: label)
        } else {
            self.init(action: destructiveAction, label: label)
        }
    }
}
```

What do you think?

Reply:
Yes, good idea 👍  Only instead of creating new init, I decided to go with a view class because if we use your provided snippet, we wouldn't be able to use  `Button { } label: { ... }`, but we would always have to specify `action` parameter like this  `Button( action: { } label: {...})`. 

```
struct DestructiveButton<Label>: View where Label : View {
    var action: () -> Void
    var label: () -> Label

    var body: some View {
        if #available(iOS 15, *) {
            return Button(role: .destructive, action: action, label: label)
        } else {
            return Button(action: action, label: label)
        }
    }
}
```

Also, I noticed that in some places where you use `destructive` button, you use different icons for iOS 15, is it intentional?

Reply:
Yeah, you could have used the Button.init with just one trailing closure like this:
```swift
Button(destructiveAction: { }) {
} 
```
which isn't that bad for me.
But I'm ok with your proposal as well.

If you are referring the button in `UserSessionOverview` for the logout, I actually copied It from `UserSessionsOverview` since I wasn't unsure about the requirements.
Probably they are leftovers so I would encourage you to cleanup also these after you verify it.